### PR TITLE
On the way to v1.3

### PR DESCRIPTION
**Pull request by VikingScientist**
Fixes several bugs in addition to add a convenience class for integration with nutils

### CODE REVIEW

File: splipy/surface_factory.py [hunk 1]
```diff
@@ -440,9 +440,9 @@ def finitestrain_patch(bottom, right, top, left):
     # create an initial mesh (correct corners) which we will morph into the right one
     p1 = bottom.order(0)
     p2 = left.order(0)
+    p  = int(max(p1,p2))
```
Comment (TheBB) [inline splipy/surface_factory.py]: Shouldn't need `int` here.

File: splipy/curve_factory.py [hunk 2]
```diff
@@ -191,38 +191,6 @@ def ellipse(r1=1, r2=1, center=(0,0,0), normal=(0,0,1), type='p2C0', xaxis=(1,0,
     result.rotate(rotate_local_x_axis(xaxis, normal))
     return flip_and_move_plane_geometry(result, center, normal)
 
-def ellipse(r1=1, r2=1, center=(0,0,0), normal=(0,0,1), type='p2C0', xaxis=(1,0,0)):
```
Comment (TheBB) [inline splipy/curve_factory.py:194]: Fixed ellipse factory by deleting it? Or am I missing something?

File: splipy/utils/nutils.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+from splipy import Curve, Surface, Volume
+import numpy as np
+
+
+def splipy_to_nutils(spline):
+    """ Return controlpoints according to nutils ordering """
+    n = len(spline)
+    dim = spline.dimension
+    if(type(spline) == Curve):
```
Comment (TheBB) [inline splipy/utils/nutils.py]: Probably better: `isinstance(spline, Curve)`

Also, parens around if-conditions? What is this, C? :smile:

**Review by TheBB — CHANGES_REQUESTED** [review]

**Comment by TheBB** [comment]
I set travis to only build if `.travis.yml` is present, so it shouldn't bother you for now.

File: splipy/curve_factory.py (hunk 2 above)
Comment (VikingScientist) [reply splipy/curve_factory.py:194]: There were two different methods with the same name and signature and no unit tests.

Comment (TheBB) [reply splipy/curve_factory.py:194]: Okay :+1:

File: splipy/utils/nutils.py (hunk 3 above)
Comment (VikingScientist) [reply splipy/utils/nutils.py]: That is better yes. Will fix

Comment (VikingScientist) [reply splipy/utils/nutils.py]: That is better, yes. Will fix it.

**Comment by VikingScientist** [comment]
Fixes as requested. If all looks okay, then it's
```
git rebase -i HEAD~7 --autosquash
```
and merge

**Review by TheBB — APPROVED** [review]

**Comment by TheBB** [comment]
Thanks, merged manually.

**Pull request closed without merging**